### PR TITLE
Ci script

### PR DESCRIPTION
**Pull request by MasoudAsadzade**
I got some surprises from ptpLib compilation into that of SMTS, so then I extended llvm/debug of ptpLib to match that of SMTS.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -40,7 +40,7 @@ jobs:
                 command: ./ci/run_ci_commands.sh
                 environment:
                     CMAKE_BUILD_TYPE: Debug
-                    FLAGS: -Wall -Wextra -Werror -fsanitize=signed-integer-overflow,address,undefined
+                    FLAGS: -Wall -Werror=sign-compare -Wunused-lambda-capture -fsanitize=signed-integer-overflow,address,undefined
```
Comment (blishko) [inline .circleci/config.yml:43]: Why did you remove `-Wextra` and `-Werror`?

Comment (MasoudAsadzade) [reply .circleci/config.yml:43]: Here for "Debug build llvm" we have to flags, FLAG and SFLAG:
```
FLAGS: -Wall -Werror=sign-compare -Wunused-lambda-capture -fsanitize=signed-integer-overflow,address,undefined
SFLAGS: -Wall -Wextra -Werror -fsanitize=signed-integer-overflow,thread,undefined
```
Link:
https://github.com/usi-verification-and-security/ptplib/blob/a53b7f174f25e7150634d8cc3a7a2e74c3b8abce/.circleci/config.yml#L44

I thought maybe it's an unnecessary check

Comment (aehyvari) [reply .circleci/config.yml:43]: Looks like SFLAGS is used for compiling protocol example and FLAGS for the rest. I think we should have strict checks everywhere to catch problems early, although checking for race conditions in sequential code makes little sense.

I thought that the PR makes the checks stricter, based on the description.

Comment (blishko) [reply .circleci/config.yml:43]: I don't know about the sanitizer, but we should keep the `-Wextra`, `-Werror` compiler flags.
